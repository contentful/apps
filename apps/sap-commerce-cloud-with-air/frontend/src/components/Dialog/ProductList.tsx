import React from 'react';
import { DialogExtensionSDK } from '@contentful/app-sdk';
import { CheckBoxFn, Product } from '../../interfaces';
import { Button, Checkbox, TableCell, TableRow } from '@contentful/forma-36-react-components';
import get from 'lodash/get';

interface Props {
  sdk: DialogExtensionSDK;
  products: Product[];
  selectedProducts: string[];
  baseSite: string;
  checkboxFn: CheckBoxFn;
}

export class ProductList extends React.Component<Props> {
  selectButtonClickEvent(sku: string) {
    const apiEndpoint = get(this.props.sdk.parameters.invocation, 'apiEndpoint', '');
    this.props.sdk.close([`${apiEndpoint}/occ/v2/${this.props.baseSite}/products/${sku}`]);
  }

  render() {
    const isFieldTypeSymbol =
      (get(this.props.sdk.parameters.invocation, 'fieldType', '') as string) === 'Symbol';

    return (
      <>
        {this.props.products.map((product, index) => {
          const checkboxValue = this.props.selectedProducts.includes(
            this.props.baseSite + ':' + product.sku
          );
          return (
            <TableRow key={product.sku}>
              <TableCell>
                {isFieldTypeSymbol ? (
                  <Button
                    buttonType="primary"
                    icon="Done"
                    onClick={() => this.selectButtonClickEvent(product.sku)}>
                    Select
                  </Button>
                ) : (
                  <Checkbox
                    labelText={''}
                    id={product.sku}
                    defaultChecked={checkboxValue}
                    onChange={this.props.checkboxFn}
                  />
                )}
              </TableCell>
              <TableCell>{product.sku}</TableCell>
              <TableCell>{product.name}</TableCell>
              <TableCell>
                <img src={product.image} alt="product" />
              </TableCell>
            </TableRow>
          );
        })}
      </>
    );
  }
}
