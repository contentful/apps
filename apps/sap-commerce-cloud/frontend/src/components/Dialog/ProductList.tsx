import { DialogExtensionSDK } from '@contentful/app-sdk';
import { CheckBoxFn, Product } from '../../interfaces';
import { Checkbox, IconButton, TableCell, TableRow } from '@contentful/f36-components';
import get from 'lodash/get';
import { formatProductUrl } from '../../utils';
import { DoneIcon } from '@contentful/f36-icons';

interface Props {
  sdk: DialogExtensionSDK;
  products: Product[];
  selectedProducts: string[];
  baseSite: string;
  checkboxFn: CheckBoxFn;
}

export function ProductList(props: Props) {
  const selectButtonClickEvent = (sku: string) => {
    const apiEndpoint = get(props.sdk.parameters.invocation, 'apiEndpoint', '') as string;
    props.sdk.close([formatProductUrl(apiEndpoint, props.baseSite, sku)]);
  };

  const isFieldTypeSymbol =
    (get(props.sdk.parameters.invocation, 'fieldType', '') as string) === 'Symbol';
  return (
    <>
      {props.products.map((product, index) => {
        const checkboxValue = props.selectedProducts.includes(props.baseSite + ':' + product.sku);
        return (
          <TableRow key={product.sku}>
            <TableCell>
              {isFieldTypeSymbol ? (
                <IconButton
                  variant="primary"
                  icon={<DoneIcon />}
                  onClick={() => selectButtonClickEvent(product.sku)}
                  aria-label="Select">
                  Select
                </IconButton>
              ) : (
                <Checkbox
                  id={product.sku}
                  defaultChecked={checkboxValue}
                  onChange={props.checkboxFn}
                />
              )}
            </TableCell>
            <TableCell>{product.sku}</TableCell>
            <TableCell>{product.name}</TableCell>
            <TableCell>
              <img src={product.image} alt="product" />
            </TableCell>
          </TableRow>
        );
      })}
    </>
  );
}
