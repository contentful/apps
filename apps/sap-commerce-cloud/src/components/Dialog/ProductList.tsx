import React from 'react';
import { DialogExtensionSDK } from '@contentful/app-sdk';
import { CheckBoxFn, Product } from '../../interfaces';
import { Button, Checkbox, TableCell, TableRow } from '@contentful/forma-36-react-components';
import get from 'lodash/get';

interface Props {
  sdk: DialogExtensionSDK;
  products: Product[];
  selectedProducts: string[];
  baseSite: string;
  checkboxFn: CheckBoxFn;
}

export class ProductList extends React.Component<Props> {
  selectButtonClickEvent(sku: string) {
    const skuId = this.props.baseSite + ':' + sku;
    this.props.sdk.close([skuId]);
  }

  render() {
    const isFieldTypeSymbol =
      (get(this.props.sdk.parameters.invocation, 'fieldType', '') as string) === 'Symbol';

    return (
      <>
        {this.props.products.map((product, index) => {
          const checkboxValue = this.props.selectedProducts.includes(
            this.props.baseSite + ':' + product.sku
          );
          return (
            <TableRow key={product.sku}>
              <TableCell>
                {isFieldTypeSymbol ? (
                  <Button
                    buttonType="primary"
                    icon="Done"
                    onClick={() => this.selectButtonClickEvent(product.sku)}>
                    Select
                  </Button>
                ) : (
                  <Checkbox
                    labelText={''}
                    id={product.sku}
                    defaultChecked={checkboxValue}
                    onChange={this.props.checkboxFn}
                  />
                )}
              </TableCell>
              <TableCell>{product.sku}</TableCell>
              <TableCell>{product.name}</TableCell>
              <TableCell>
                <img src={product.image} />
              </TableCell>
            </TableRow>
          );
        })}
      </>
    );
  }
}
