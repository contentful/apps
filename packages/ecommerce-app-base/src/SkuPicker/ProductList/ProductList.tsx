import React from 'react';
import tokens from '@contentful/f36-tokens';
import { css } from '@emotion/css';
import { Product } from '../../types';
import { ProductListItem } from './ProductListItem';

export interface Props {
  products: Product[];
  selectProduct: (sku: string) => void;
  selectedSKUs: string[];
}

const styles = {
  productList: css({
    display: 'flex',
    flexWrap: 'wrap',
    marginLeft: `-${tokens.spacingS}`,
    marginRight: `-${tokens.spacingS}`,
  }),
};

export function ProductList({ selectProduct, selectedSKUs, products }: Props) {
  return (
    <div className={styles.productList}>
      {products.map((product) => (
        <ProductListItem
          key={product.id}
          product={product}
          isSelected={selectedSKUs.includes(product.sku)}
          selectProduct={selectProduct}
        />
      ))}
    </div>
  );
}
