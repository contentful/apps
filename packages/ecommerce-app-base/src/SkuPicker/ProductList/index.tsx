import React from 'react';
import tokens from '@contentful/forma-36-tokens';
import { css } from '@emotion/css';
import { Product } from '../../interfaces';
import { ProductListItem } from './ProductListItem';

export interface Props {
  products: Product[];
  selectProduct: (sku: string) => void;
  selectedSKUs: string[];
  skuLabel?: string;
  readableIdentifierLabel?: string;
}

const styles = {
  productList: css({
    display: 'flex',
    flexWrap: 'wrap',
    marginLeft: `-${tokens.spacingS}`,
    marginRight: `-${tokens.spacingS}`
  })
};

export const ProductList = ({
  selectProduct,
  selectedSKUs,
  products,
  skuLabel,
  readableIdentifierLabel,
}: Props) => (
  <div className={styles.productList}>
    {products.map(product => (
      <ProductListItem
        key={product.id}
        product={product}
        isSelected={selectedSKUs.includes(product.sku)}
        selectProduct={selectProduct}
        skuLabel={skuLabel}
        readableIdentifierLabel={readableIdentifierLabel}
      />
    ))}
  </div>
);
