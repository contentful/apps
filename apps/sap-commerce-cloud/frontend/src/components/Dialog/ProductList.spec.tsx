import { render, screen } from '@testing-library/react';
import { ProductList } from './ProductList';
import { vi } from 'vitest';
import { DialogExtensionSDK } from '@contentful/app-sdk';

describe('ProductList', () => {
  it('renders', () => {
    const { container } = render(
      <ProductList
        {...{
          sdk: {
            parameters: {
              invocation: {},
            },
          } as unknown as DialogExtensionSDK,
          products: [],
          selectedProducts: [],
          baseSite: '',
          checkboxFn: vi.fn(),
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it('selectButtonClickEvent', async () => {
    const close = vi.fn();
    const { container } = render(
      <ProductList
        {...{
          sdk: {
            parameters: {
              invocation: {
                fieldType: 'Symbol',
              },
            },
            close,
          } as unknown as DialogExtensionSDK,
          products: [{ sku: '123', image: '', id: '', name: '', productUrl: '' }],
          selectedProducts: [],
          baseSite: '',
          checkboxFn: vi.fn(),
        }}
      />
    );
    const selectButton = await screen.findByLabelText('Select');
    selectButton.click();
    expect(close).toHaveBeenCalled();
  });
});
