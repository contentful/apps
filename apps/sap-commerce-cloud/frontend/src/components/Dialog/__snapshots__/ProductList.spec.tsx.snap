// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductList > renders 1`] = `<div />`;
